pkg/commands/workflows/remove_comments.go

```diff
@@ -55,6 +56,8 @@ func (c *RemoveGuardianCommentsCommand) Flags() *cli.FlagSet {
 }
 
 func (c *RemoveGuardianCommentsCommand) Run(ctx context.Context, args []string) error {
+	metricswrap.WriteMetric(ctx, "command_workflows_remove_guardian_comments", 1)
```

Comment:
I like this developer API better. As a follow-up, I wonder if we could automatically export these metrics magically in the CLI package.

Reply:
That's an interesting thought. Current design of metrics does still require allow listing each metric so you would still have an extra step.

As for how to expose in CLI package, I can see a few options.
1. Have command factory inject the command(s) that caused this command to run into the cli.Command interface. []string. For example run would just be ["run"] while iam cleanup would be ["iam", "cleanup"]. 

The main drawback to which approach is if you have multiple possible paths to the same Command (for example `abc` dropped the `templates` in `abc templates render`, but had to keep it as an option for backwards compatibility.

2. Have cli.Command expose its type somehow. Either with an explicit field or getting struct name via reflection.

Refelction feels like a bit of a smell, so probably not the best choice.



As far as dispatching the metrics go, if you don't want to be explicit, I would think you would probably want something more generic, like adding an arbitrary function to the context which gets called before Run().

Reply:
I was thinking more generic actually. The CLI can introduce things like "pre" and "post" actions (e.g. `BeforeRun`, `AfterRun`), and the main registration could hook into that. We already do this a bit with `AfterParse` FWIW, but you could register metrics as a hook.